lib/catalogx_client/product.rb

```diff
@@ -1,112 +1,117 @@
+          params = { overwrite: overwrite, testing: true }
           resp = handle_request(url_path, :post, query_params: params, body: {products: products})
           api = overwrite ? 'bulk_create' : 'bulk_update'
           CATALOGX_LOGGER.info("service=catalogx|api=#{api}|response=#{resp.to_s}")
         rescue => ex
           CATALOGX_LOGGER.error("#{ex}")
-          catalogx_client_statsd_exception('batch_upsert', @migration_status)
+          catalogx_client_statsd_exception('batch_upsert', account_migration_status, account_uuid)
         end
 
-        resp = uts_batch(products, overwrite)
+        resp = uts_batch(products, account_uuid, overwrite, caller_ctx)
         api = overwrite ? 'bulk_create' : 'bulk_update'
         CATALOGX_LOGGER.info("service=uts|api=#{api}|response=#{resp.to_s}")
         resp
-      elsif @migration_status == 'complete'
+      elsif account_migration_status == 'complete'
         params = { overwrite: overwrite }
         resp = handle_request(url_path, :post, query_params: params, body: {products: products})
       else
-        uts_batch(products, overwrite)
+        begin
+          uts_batch(products, account_uuid, overwrite, caller_ctx)
+        rescue  => ex
+          CATALOGX_LOGGER.error("#{ex}")
+          catalogx_client_statsd_exception('batch_upsert', account_migration_status, account_uuid)
+        end
       end
     end
 
-    def set_out_of_stock(catalog_uuid)
-      url_path = "accounts/#{@account_uuid}/products/set_out_of_stock"
-      if @migration_status == 'log_responses'
+    def self.set_out_of_stock(catalog_uuid, account_uuid, caller_ctx)
+      account_migration_status = get_migration_status(account_uuid)
+      url_path = "accounts/#{account_uuid}/products/set_out_of_stock"
+
+      if account_migration_status == 'log_responses'
         begin
-          params = { overwrite: overwrite }
+          params = { testing: true }
           body =  {"catalog_uuid" => catalog_uuid}
           resp = handle_request(url_path, :post, query_params: params, body: body)
           CATALOGX_LOGGER.info("service=catalogx|api=set_out_of_stock|response=#{resp.to_s}")
         rescue => ex
           CATALOGX_LOGGER.error("#{ex}")
-          catalogx_client_statsd_exception('set_out_of_stock', @migration_status)
+          catalogx_client_statsd_exception('set_out_of_stock', account_migration_status, account_uuid)
         end
 
-        resp = uts_set_stock(catalog_uuid)
+        resp = uts_set_out_of_stock(catalog_uuid, caller_ctx, account_uuid)
         CATALOGX_LOGGER.info("service=uts|api=set_out_of_stock|response=#{resp.to_s}")
         resp
-      elsif @migration_status == 'complete'
-        params = { overwrite: overwrite }
+      elsif account_migration_status == 'complete'
         body = {"catalog_uuid" => catalog_uuid}
-        handle_request(url_path, :post, query_params: params, body: body)
+        handle_request(url_path, :post, query_params: {}, body: body)
```

Comment:
params should be passed here, right?

Reply:
no, the overwrite param wouldn't matter either way since it only affects the `extended_attributes` - while this action of setting a catalog out of stock just changes the `availability` of a product, so either way it would be fine.